examples/tutorials/matmul/int4_weight_decompression_cmnts.cpp

```diff
@@ -0,0 +1,265 @@
+/*******************************************************************************
+ * Copyright 2023-2024 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+/// @example int4_weights_decompression.cpp
+/// > Annotated version: @ref int4_weights_decompression.cpp
+///
+/// @page int4_weights_decompression
+/// C++ API example demonstrating how one can use
+/// [MatMul](@ref dev_guide_matmul) with int4 compressed weights.
+///
+/// Concepts:
+/// - AWQ (activation-aware quantization)
+///   - Scales: dnnl::primitive_attr::set_scales()
+///   - Zero points: dnnl::primitive_attr::set_zero_points()
+/// - [Operation fusion](@ref dev_guide_attributes_post_ops)
+/// - Create primitive once, use multiple times
+/// - Weights pre-packing: use #dnnl::memory::format_tag::any
+///
+/// @page int4_weights_decompression_matmul_cpp MatMul Tutorial: weights
+/// decompression
+/// @copydetails int4_weights_decompression_matmul_cpp
+///
+/// Assumptions:
+/// 1. The shape of the weights (matrix \f$B(K, N)\f$) is known in advance, the
+///    data type is `int4` and shifted from 0 (i.e. the zero point is not 0).
+/// 2. The source matrix \f$A\f$ and destination matrix \f$C\f$ have floating
+///    point data type.
+/// 3. Scaling (re-quantization) factor specified at run-time only.
+///
+/// Since the shape of weights is known in advance, the MatMul weights can be
+/// created with format tag #dnnl::memory::format_tag::any to enable the library
+/// to choose the most appropriate layout for best performance.
+///
+/// @warning
+/// The format tag #dnnl::memory::format_tag::any doesn't work for memory
+/// descriptors that have one or more unknown dimensions and/or strides.
+///
+/// @include weights_decompression_matmul.cpp
```

Comment:
This line needs update. It's not weights_decompression_matmul.cpp

Reply:
renamed the cpp file